Let ListImages() return a string instead printing to stdout

miha-plesko opened the pull request:
With this commit we modify `ListImages` so that it now returns a string, which opens an opportunity to provide unit tests for it. We grab the opportunity and actually provide the unit tests. Having tests in place, we're confident enough to refactor the body of ListImages method that was implemented in a strange way and thus hardly understandable.

Depends on https://github.com/mikelangelo-project/capstan/pull/56

gberginc commented:
@miha-plesko please rebase with #56.

miha-plesko commented:
Rebased, repushed, green.

gberginc commented on util/s3_repository.go:
```diff
@@ -52,22 +52,24 @@ func FileInfoHeader() string {
 
 func (f *FileInfo) String() string {
-	return fmt.Sprintf("%-50s %-50s %-15s %-20s", f.Namespace+"/"+f.Name, f.Description, f.Version, f.Created)
+	name := strings.TrimLeft(f.Namespace+"/"+f.Name, "/")
```
Please, add a comment explaining that you are doing this because `f.Namespace` can be empty and are thus making sure that you don't get image names like `/myimage`.

miha-plesko commented:
I feel you - I had to test myself again if it works prior opening the PR, since I've implemented it long time ago and looking at the diff I was like "whuuuuuuuuut?!?!" 😄 

I've added the comment that you asked for, please see.
